src/utl/src/timer.cpp

```diff
@@ -74,4 +78,71 @@ DebugScopedTimer::~DebugScopedTimer()
   debugPrint(logger_, tool_, group_.c_str(), level_, msg_, *this);
 }
 
+ScopedStatistics::ScopedStatistics(utl::Logger* logger, std::string msg)
+    : Timer(),
+      msg_(std::move(msg)),
+      start_rsz_(getStartRSZ()),
+      start_vsz_(getStartVSZ()),
+      logger_(logger)
+{
+}
+
+size_t ScopedStatistics::getMemoryUsage(const char* tag)
+{
+  FILE* file = fopen("/proc/self/status", "r");
+  if (file == nullptr) {
+    perror("Failed to open /proc/self/status");
+    exit(EXIT_FAILURE);
```

Comment:
This is too harsh a form of error handling.  Issuing a logger warning is sufficient.